src/lang-puppet.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright (C) 2016 Jorge Morgado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Registers a language handler for Puppet.
+ *
+ * @author jorge at morgado dot ch
+ */
+
+// Falls back to 'plain text' for stylesheets that don't style opn and clo.
+var PR_OPEN  = 'opn pln';
+var PR_CLOSE = 'clo pln';
+
+// Falls back to 'attribute name' for stylesheets that don't style fun.
+var PR_FUNCTION = 'fun atn';
+// Falls back to 'literal' for stylesheets that don't style var.
+var PR_VARIABLE = 'var lit';
+
+PR['registerLangHandler'](
+    PR['createSimpleLexer'](
+        [
+          // Open and close
+          [PR_OPEN,              /^[\(\{\[]+/, null, '([{'],
+          [PR_CLOSE,             /^[\)\}\]]+/, null, ')]}'],
+
```

Comment:
There are additional Open/Close pairs:
* `|` delimiters around lambda parameters - e.g. `$arr.each | $x | { }
* "spaceship-operators" i.e. `<| |>`, and `<<| |>>` for collection operations
* Heredoc @(TAG) TAG sequence (complex to lex)